core/src/main/java/org/opensearch/sql/storage/Table.java

```diff
@@ -92,4 +92,9 @@ default TableWriteBuilder createWriteBuilder(LogicalWrite plan) {
   default StreamingSource asStreamingSource() {
     throw new UnsupportedOperationException();
   }
+
+  default TableScanBuilder createPagedScanBuilder(int pageSize) {
```

Comment:
we should consider reuse createScanBuilder(), it is not clear on the definition of new interface.

Reply:
Is it a blocker for now or it can be done later?

Reply:
if blocker means sepereate execution path and duplicate code which increase maintaince effort, yes, it is blocker.
if blocker means quick feature delivery, no, it is not blcoker.

Reply:
@penghuo Merging paged and non-paged scan builders leads to merging OpenSearchIndexScan and OpenSearchPagedIndexScan. That will take some care -- OpenSearchIndexScan already issues scroll requests in some cases for non-paged requests.

We are looking into this right now. It can be included in [follow-up work](https://github.com/opensearch-project/sql/pull/1500) on pagination.